providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/common/ChannelBuilder.java

```diff
@@ -78,6 +94,46 @@ public static ManagedChannel nettyChannel(final FlagdOptions options) {
             SslConfigException sslConfigException = new SslConfigException("Error with SSL configuration.");
             sslConfigException.initCause(ssle);
             throw sslConfigException;
+        } catch (IllegalArgumentException argumentException) {
+            GenericConfigException genericConfigException = new GenericConfigException(
+                    "Error with gRPC target string configuration");
+            genericConfigException.initCause(argumentException);
+            throw genericConfigException;
+        }
+    }
+
+    private static  boolean isValidTargetUri(String targetUri) {
+        if (targetUri == null) {
+            return false;
+        }
+
+        try {
+            final String scheme = new URI(targetUri).getScheme();
+            if (scheme.equals("envoy") || scheme.equals("dns") || scheme.equals("xds") || scheme.equals("uds")) {
```

Comment:
[suggestion] should we maybe put those strings into an enum, or maybe there is an existing enum - i am not a big fan of strings, and specially if they are duplicated.

Reply:
ack

Reply:
@aepfli thanks for the suggestion .. added a new enum class `SupportedScheme` and replace the string with enum